src/detect.ts

```diff
@@ -43,7 +43,16 @@ export async function detect({ autoInstall, programmatic, cwd }: DetectOptions =
         process.exit(1)
     }
 
-    await ezspawn(`npm i -g ${agent.split('@')[0]}${version ? `@${version}` : ''}`, { stdio: 'inherit', cwd })
+    await x(
+      'npm',
+      ['i', '-g', `${agent.split('@')[0]}${version ? `@${version}` : ''}`],
+      {
+        nodeOptions: {
+          stdio: 'inherit',
+          cwd,
+        },
+      },
```

Comment:
to keep the behavior of `ezspawn` we need the new `throwOnError` added to 0.3.0
```suggestion
        },
        throwOnError: true,
      },
```